l10n_it_fiscalcode/model/res_partner.py

```diff
@@ -19,29 +19,42 @@
 #
 ##############################################################################
 
-from openerp import models, fields, api
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
+import logging
 
+_logger = logging.getLogger(__name__)
+
+try:
+    import codicefiscale
+except ImportError:
+    _logger.warning(
+        'codicefiscale library not found. '
+        'If you plan to use it, please install the codicefiscale library '
+        'from https://pypi.python.org/pypi/codicefiscale')
 
-class res_partner(models.Model):
-    _inherit = 'res.partner'
 
-    @api.multi
-    def check_fiscalcode(self):
-        for partner in self:
-            if not partner.fiscalcode:
-                return True
-            elif len(partner.fiscalcode) != 16 and partner.individual:
-                return False
-            else:
-                return True
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
 
     fiscalcode = fields.Char(
-        'Fiscal Code', size=16, help="Italian Fiscal Code")
+        string='Fiscal Code', size=16, help="Italian Fiscal Code")
     individual = fields.Boolean(
-        'Individual', default=False,
+        string='Individual', default=False,
         help="If checked the C.F. is referred to a Individual Person")
 
-    _constraints = [
-        (check_fiscalcode,
-         "The fiscal code doesn't seem to be correct.", ["fiscalcode"])
-    ]
+    @api.one
```

Comment:
Hi @gigidn  thanks

api.one in 9.0 is deprecated as a described [here](https://github.com/odoo/odoo/blob/9.0/openerp/api.py#L405)